brambling/views/orders.py

```diff
@@ -247,6 +247,22 @@ def dispatch(self, request, *args, **kwargs):
                                        organization__slug=kwargs['organization_slug'])
         if not self.event.viewable_by(request.user):
             raise Http404
+        if 'code' in kwargs:
+            # If an anonymous user is trying to access an anonymous
+            # order without a session, redirect them to the login
+            # page.
+            matching_order = Order.objects.filter(event=self.event,
+                                                  person__isnull=True,
+                                                  code=kwargs['code']).exists()
+            if (matching_order and not request.user.is_authenticated() and
+                not Order.objects._get_session_code(request, self.event)):
+                url = reverse('login')
```

Comment:
I think I might prefer in this case to display a helpful message, as well as either a login or signup form depending on whether the account already exists. (Rather than just redirecting to login.) At a bare minimum we should add a message (using `django.contrib.messages`) explaining wtf is going on.